plugins/samples/check_pii/plugin.cc

```diff
@@ -0,0 +1,84 @@
+// Copyright 2024 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+// [START serviceextensions_plugin_check_pii]
+#include "proxy_wasm_intrinsics.h"
+#include "re2/re2.h"
+
+class MyRootContext : public RootContext {
+ public:
+  explicit MyRootContext(uint32_t id, std::string_view root_id)
+      : RootContext(id, root_id) {}
+
+  bool onConfigure(size_t) override {
+    // Credit card numbers in a 19 character hyphenated format.
+    // Compile the regex expression at plugin setup time.
+    card_match.emplace("\\d{4}-\\d{4}-\\d{4}-(\\d{4})");
+    return card_match->ok();
+  }
+
+  std::optional<re2::RE2> card_match;
+};
+
+// Checks the response http headers, and the response body for the presence of
+// credit card numbers. Mask the initial numbers case found.
+class MyHttpContext : public Context {
+ public:
+  explicit MyHttpContext(uint32_t id, RootContext* root)
+      : Context(id, root), root_(static_cast<MyRootContext*>(root)) {}
+
+  FilterHeadersStatus onResponseHeaders(uint32_t headers,
+                                        bool end_of_stream) override {
+    const auto pii_header = getResponseHeader("google-run-pii-check");
+    if (pii_header && pii_header->view() == "true") {
+      check_body_ = true;
+      const auto result = getResponseHeaderPairs();
```

Comment:
nit: I feel like it might help readability to expand some of these `auto` types--in the Google C++ style guide, the general guidance is to expand `auto` any place where the type of the variable is not readily apparent from the rest of the expression.

That said, a lot of our existing examples use `auto` fairly liberally, so maybe that argues for sticking with `auto` here for consistency, and doing a general pass to explicitly specify more types in a separate PR. So I'm not suggesting a change here, but just a heads up that I may send a future PR to spell out types more explicitly in this and other examples.

Reply:
Agreed... So, I wont change here and wait for your PR to change it in more places. Tks